src/condor_daemon_core.V6/daemon_core.cpp

```diff
@@ -1695,44 +1683,42 @@ int DaemonCore::Register_Socket(Stream *iosock, const char* iosock_descrip,
 
 
 	// Verify that this socket has not already been registered
-	// Since we are scanning the entire table to do this (change this someday to a hash!),
-	// at the same time update our nRegisteredSocks count by initializing it
-	// to the number of slots (nSock) and then subtracting out the number of slots
-	// not in use.
-	nRegisteredSocks = nSock;
+	// Since we are scanning the entire table to do this 
+	// at the same time update our nRegisteredSocks count
+	nRegisteredSocks = 0;
 	int fd_to_register = ((Sock *)iosock)->get_file_desc();
 	bool duplicate_found = false;
-	for ( j=0; j < nSock; j++ )
+	for ( size_t j=0; j < sockTable.size(); j++ )
 	{		
-		if ( (*sockTable)[j].iosock == iosock ) {
+		if ( sockTable[j].iosock == iosock ) {
 			i = j;
 			duplicate_found = true;
         }
 
 		// fd may be -1 if doing a "fake" registration: reverse_connect_pending
 		// so do not require uniqueness of fd in that case
-		if ( (*sockTable)[j].iosock && fd_to_register != -1 ) {
-			if ( ((Sock *)(*sockTable)[j].iosock)->get_file_desc() ==
+		if ( sockTable[j].iosock && fd_to_register != -1 ) {
+			if ( ((Sock *)sockTable[j].iosock)->get_file_desc() ==
 								fd_to_register ) {
 				i = j;
 				duplicate_found = true;
 			}
 		}
 
 		// check if slot empty or available
-		if ( ((*sockTable)[j].iosock == NULL) ||  // slot is empty
-			 ((*sockTable)[j].remove_asap &&	   // slot available
-			           (*sockTable)[j].servicing_tid==0 ) ) 
+		if ((sockTable[j].iosock != nullptr) &&  // slot is empty
+			 !((sockTable[j].remove_asap &&	   // slot available
```

Comment:
I suggest rewording the comments. Maybe "is slot empty?" and "is slot available?".